include/accessor.h

```diff
@@ -307,61 +354,62 @@ class accessor<DataT, Dims, Mode, target::host_task> : public detail::accessor_b
 		} else {
-			if constexpr(Target == target::host_task) {
-				if(detail::get_handler_execution_target(cgh) != detail::execution_target::host) {
-					throw std::runtime_error(
-					    "Calling accessor constructor with host_buffer target is only allowed in host tasks."
-					    "If you want to access this buffer from within a parallel_for task, please specialize the call using one of the non host tags");
-				}
-				auto& live_cgh = dynamic_cast<detail::live_pass_host_handler&>(cgh);
-				// It's difficult to figure out which stored range mapper corresponds to this constructor call, which is why we just call the raw mapper
-				// manually.
-				const auto sr = live_cgh.apply_range_mapper<Dims>(rmfn, buff.get_range());
-				const auto access_info =
-				    detail::runtime::get_instance().get_buffer_manager().access_host_buffer<DataT, Dims>(detail::get_buffer_id(buff), Mode, sr);
-
-				m_mapped_subrange = sr;
-				m_host_ptr = static_cast<DataT*>(access_info.ptr);
-				m_index_offset = detail::id_cast<Dims>(access_info.backing_buffer_offset);
-				m_buffer_range = detail::range_cast<Dims>(access_info.backing_buffer_range);
-				m_virtual_buffer_range = buff.get_range();
+			if(detail::get_handler_execution_target(cgh) != detail::execution_target::host) {
+				throw std::runtime_error(
+				    "Calling accessor constructor with host_buffer target is only allowed in host tasks."
+				    "If you want to access this buffer from within a parallel_for task, please specialize the call using one of the non host tags");
 			}
+			auto& live_cgh = dynamic_cast<detail::live_pass_host_handler&>(cgh);
+			// It's difficult to figure out which stored range mapper corresponds to this constructor call, which is why we just call the raw mapper
+			// manually.
+			const auto sr = live_cgh.apply_range_mapper<Dims>(rmfn, buff.get_range());
+			auto access_info = detail::runtime::get_instance().get_buffer_manager().access_host_buffer<DataT, Dims>(detail::get_buffer_id(buff), Mode, sr);
+
+			m_mapped_subrange = sr;
+			m_host_ptr = static_cast<DataT*>(access_info.ptr);
+			m_index_offset = detail::id_cast<Dims>(access_info.backing_buffer_offset);
+			m_buffer_range = detail::range_cast<Dims>(access_info.backing_buffer_range);
+			m_virtual_buffer_range = buff.get_range();
 		}
 	}
 
-	template <typename Functor, typename TagT>
-	accessor(const buffer<DataT, Dims>& buff, handler& cgh, Functor rmfn, TagT tag) : accessor(buff, cgh, rmfn) {}
+	template <typename Functor, access_mode TagModeNoInit>
+	accessor(const buffer<DataT, Dims>& buff, handler& cgh, const Functor& rmfn, const detail::access_tag<Mode, TagModeNoInit, target::host_task> /* tag */)
+	    : accessor(buff, cgh, rmfn) {}
 
 	/**
 	 * TODO: As of ComputeCpp 2.5.0 they do not support no_init prop, hence this constructor is needed along with discard deduction guide.
 	 *    but once they do this should be replace for a constructor that takes a prop list as an argument.
 	 */
```

Comment:
We no longer support ComputeCpp as of 68367dd, can we just drop this?

Reply:
We can re-think a large part of our `access_mode` logic now that USM buffers are merged and we do not need to pass on the `no_init` information through `access_mode::discard_write` anymore. This would also allow us to support accessor property lists for real (if that is even desirable). Since this line is not directly affected by this PR I would vote to postpone this though.